### PR TITLE
Update DataFusion proto

### PR DESCRIPTION
**Pull request by Dandandan**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by thinkharderdev — APPROVED** [review]

File: ballista/core/proto/datafusion.proto [hunk 1]
```diff
@@ -852,47 +988,54 @@ message ScalarValue {
...
-    Timestamp TIMESTAMP = 20;
-    TimeUnit TIME32 = 21;
-    TimeUnit TIME64 = 22;
-    IntervalUnit INTERVAL = 23;
-    Decimal DECIMAL = 24;
+    Timestamp TIMESTAMP = 20 ;
+    TimeUnit TIME32 = 21 ;
+    TimeUnit TIME64 = 22 ;
+    IntervalUnit INTERVAL = 23 ;
+    Decimal DECIMAL = 24 ;
```
Comment (ch-sc) [inline ballista/core/proto/datafusion.proto:1031]: Looks like you applied a lot of formatting in this PR. These whitespaces seem off though.

File: ballista/core/proto/datafusion.proto [hunk 2]
```diff
@@ -1225,13 +1406,32 @@ enum AggregateMode {
...
+
+message PartiallySortedPartitionSearchMode {
+  repeated uint64 columns = 6;
 }
 
 message WindowAggExecNode {
   PhysicalPlanNode input = 1;
-  repeated PhysicalExprNode window_expr = 2;
-  repeated string window_expr_name = 3;
-  Schema input_schema = 4;
```
Comment (ch-sc) [inline ballista/core/proto/datafusion.proto:1234]: Should these be marked `reserved` or were they not used before?

File: ballista/core/proto/datafusion.proto [hunk 3]
```diff
@@ -1295,18 +1507,18 @@ message RepartitionExecNode {
...
 }
 
-message ColumnIndex {
+message ColumnIndex{
   uint32 index = 1;
   JoinSide side = 2;
 }
 
-enum JoinSide {
+enum JoinSide{
```
Comment (ch-sc) [inline ballista/core/proto/datafusion.proto:1521]: Inconsistent formatting changes. There are a few more of these

Comment (Dandandan) [reply ballista/core/proto/datafusion.proto:1521]: yeah, not sure we want to fix it, as this is a literal copy from upstream datafusion

**Pull request merged**